pkg/mesh/mesh.go

```diff
@@ -469,7 +470,9 @@ func (m *Mesh) applyTopology() {
 		return
 	}
 	oldConf := wireguard.Parse(oldConfRaw)
-	updateNATEndpoints(nodes, peers, oldConf)
+	natEndpoints := updateNATEndpoints(nodes, peers, oldConf, m.logger)
+	nodes[m.hostname].DiscoveredEndpoints = natEndpoints
+	m.nodes[m.hostname].DiscoveredEndpoints = natEndpoints
```

Comment:
`nodes` is a map of pointers to Nodes, so I think updating the `nodes` map should suffice and we should not need to update `m.nodes` as well

Reply:
The `applyTopology` function starts by doing shallow copies of the Ready Nodes ...
Updating `nodes` will provide the updated endpoint to `NewTopology`.
Updating `m.nodes`  ensure that next call to `checkIn` will publish the `DiscoveredEndpoints` to k8s.

There are others way to do it, but this looked easy enough.

Reply:
Yes I definitely agree that we want both `nodes[m.hostname]` and `m.nodes[m.hostname]` to include the updated endpoints so that the `checkIn` method can update the endpoints annotation.

What I don't see is why we need to explicitly set `.DiscoveredEndpoints = natEndpoints` for both of them. Since `nodes` is a shallow copy, updating `nodes[m.hostname]` will also update `m.nodes[m.hostname]`, no?

e.g. https://play.golang.com/p/JXCE1jnUN-c

Reply:
Well the copy is not a pointer copy but a shallow copy so more like:
https://play.golang.com/p/istIdrj__0_P

for it to work I need to make the endpoint list a pointer so that the copy points to the same map
eg: https://play.golang.com/p/M2uunAQXRyI

I can do that if you think it's better that way.

Reply:
Yes, you're totally right 🤦https://github.com/squat/kilo/blob/749cdd1192a15ef5345d4172fdbe373c55097918/pkg/mesh/mesh.go#L435

This lgtm then

Reply:
Thinking about this some more, it occurs to me that the only reason we really had to do the shallow copies was to prevent the endpoints of the nodes to be overwritten by the `updateNATEndpoints` func, which used to mutate the parameters. Now that the func no longer mutates data we could probably switch from shallow copies to just copying the pointers. We can do this in a follow up however

Reply:
Done